markets/perps-market/test/integration/Position/Funding.test.ts

```diff
@@ -64,48 +70,82 @@ describe('Position - funding', () => {
   */
 
   [
-    { daysElapsed: 1, newOrderSize: bn(-60), expectedAccruedFunding: bn(-120) },
-    { daysElapsed: 2, newOrderSize: bn(15), expectedAccruedFunding: bn(-120) },
-    { daysElapsed: 2.25, newOrderSize: bn(37), expectedAccruedFunding: bn(-50.63) },
-    { daysElapsed: 4.25, newOrderSize: bn(-17), expectedAccruedFunding: bn(291.38) },
-    { daysElapsed: 4.45, newOrderSize: bn(155), expectedAccruedFunding: bn(297.98) },
-    { daysElapsed: 4.55, newOrderSize: bn(-150), expectedAccruedFunding: bn(292.88) },
-    { daysElapsed: 4.65, newOrderSize: bn(-15), expectedAccruedFunding: bn(278.78) },
-    { daysElapsed: 4.68, newOrderSize: bn(-4), expectedAccruedFunding: bn(274.63) },
-    { daysElapsed: 7.68, newOrderSize: bn(19), expectedAccruedFunding: bn(893.83) },
-  ].forEach(({ daysElapsed, newOrderSize, expectedAccruedFunding }) => {
-    describe(`after ${daysElapsed} days`, () => {
-      before('move time', async () => {
-        await fastForwardTo(
-          openPositionTime -
-            // this enables the market summary check to be as close as possible to the settlement time
-            (DEFAULT_SETTLEMENT_STRATEGY.settlementDelay - 1) + // settlement strategy delay accounted for
-            SECONDS_IN_DAY * daysElapsed,
-          provider()
-        );
-      });
+    { daysElapsed: 1, newOrderSize: bn(-60) },
+    { daysElapsed: 1, newOrderSize: bn(15) },
+    { daysElapsed: 0.25, newOrderSize: bn(37) },
+    { daysElapsed: 2, newOrderSize: bn(-17) },
+    { daysElapsed: 0.2, newOrderSize: bn(155) },
+    { daysElapsed: 0.1, newOrderSize: bn(-150) },
+    { daysElapsed: 0.1, newOrderSize: bn(-15) },
+    { daysElapsed: 0.03, newOrderSize: bn(-4) },
+    { daysElapsed: 3, newOrderSize: bn(19) },
+  ].reduce(
+    (
+      { prevFrVelocity, prevFundingRate, prevAccruedFunding, prevSkew, accDaysElapsed },
+      { daysElapsed, newOrderSize }
+    ) => {
+      accDaysElapsed += daysElapsed;
+
+      // ACCRUED FUNDING CALC ----------------------------------------------
+      const currentSkew = prevSkew.add(newOrderSize);
+      const frVelocity = wei(currentSkew).div(_SKEW_SCALE).mul(_MAX_FUNDING_VELOCITY);
+      const fundingRate = prevFrVelocity.mul(daysElapsed).add(prevFundingRate);
+      const expectedAccruedFunding = Wei.avg(wei(prevFundingRate), wei(fundingRate))
```

Comment:
I added `calcCurrentFundingVelocity` in a `helpers/funding-calcs.ts` that you could use. We could also move some of these there if we think other tests will assert funding updates

Reply:
+1